src/cc/usdt/usdt.cc

```diff
@@ -272,13 +277,41 @@ Probe *Context::get(const std::string &probe_name) {
   return nullptr;
 }
 
+Probe *Context::get(const std::string &provider_name,
+                    const std::string &probe_name) {
+  for (auto &p : probes_) {
+    if (p->provider_ == provider_name && p->name_ == probe_name)
+      return p.get();
+  }
+  return nullptr;
+}
+
 bool Context::enable_probe(const std::string &probe_name,
                            const std::string &fn_name) {
   if (pid_stat_ && pid_stat_->is_stale())
     return false;
 
-  auto p = get(probe_name);
-  return p && p->enable(fn_name);
+  // FIXME: we may have issues here if the context has two same probes's
```

Comment:
Can we also have a `provider_name` **and** `probe_name` version of `enable_probe`?

Reply:
For C++, we do not need `enable_probe(provider_name, probe_name)` yet as no C++ API will use it. For Python API, currently, all our major users, `trace.py`, `argdist.py` and `funccount.py` all use `u:<binary_path>:<probe_name>` to specify usdt, and hence, `enable_probe(probe_name)` is used.

We could have `enable_probe(provider_name, probe_name)` for some specific tools which uses usdt (e.g., tracing java, ruby and python etc.) where no user interface change is needed. This is nice to have but not required as it is a controlled environment, `enable_probe(probe_name)` should just work.

But we cannot get rid of `enable_probe(probe_name)` due to python API with `trace.py` etc. Do we want to change `trace.py` etc. to have `provider_name` name to handle the case where the same binary contains both `provider_a:probe_name` and `provider_b:probe_name`? We can discuss this separately.

Reply:
I see. For correctness probably yes, but I'd like to know what @goldshtn and @brendangregg thinks......